### DISCUSSION TITLE
Предложения различных математических подпрограмм

### DISCUSSION THREAD

**Discussion by Anonymous87549236** [pascalabcnet/pascalabcnet#2372] [Ideas]
#2370

**Comment by Anonymous87549236** [comment]
Sin дробью:

```
uses NumLibABC; 

//Источник: https://www.cyberforum.ru/cpp-beginners/thread1598455.html

begin
  while true do
  begin
    var x := Frc(ReadlnBigInteger, ReadlnBigInteger);
    var ε := Frc(1, '100000000000000000000000000000000000000000000000000000000000000000000000000000000000'.ToBigInteger);
    var sum := Frc(0);
    var n := BigInteger.One;
    var an:=x;
    while an.Abs>ε do
    begin
      sum+=an;
      n+=1;
      an*=-x*x/(2*n-1)/(2*n-2);
    end;
    Writeln(sum);
  end;
end.
```

**Comment by Anonymous87549236** [comment]
В конце концов для всех нецелых типов чисел (real, decimal, fraction и т. д.) предлагаю добавить следующее (кроме существующего):
1. В остальные типы — все совместимые математические подпрограммы, присутствующие в каком-либо одном типе, с максимальной точностью каждого типа соответственно, для fraction — с указанием точности, если число иррациональное.
2. Преобразование типов, например, fraction в decimal.
3. Возможность вывода с округлением без потери точности, например, перевод в строку с последующим округлением.
4. Все возможные тригонометрические функции.
5. Вывод по желанию в виде дроби, экспоненциальной записи и обычного числа.

**Comment by miks1965** [comment]
Fraction делать не будем.

r.ToString(2) уже есть

Округление до определенного разряда для вещественных - странная затея - всё равно точно не будет

Все возможные тригонометрические есть в классе Math
